### PR TITLE
[AI] README 업데이트 및 issue #36 해결을 위한 PR

### PR DESCRIPTION
**Pull request by 4923**
- 데이터 추가
- xgboost, ann 튜닝 결과 반영
- 모듈화 완료
- 데이터 추가: 이 커밋이 이전에 PR로 올라가지 않은것도 문제같다
- **발전량 실행파일 경로 변경** 0f22a14471a9250704504ddeec961a0d2ccdad4a
백엔드 src에서 잘못된 파일명을 부르고 있길래 변경함

### 기타
튜닝하면서 자잘하게 변형된게 있었는데 그 내용들이 전부 반영되지 않은것 같으니 한번 확인해봅시다

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # private DATA
-**/private/
+# **/private/
```
Comment (4923) [inline .gitignore:2]: #35 에서 ignore 풀어놓고 머지했던걸로 기억하는데... 🤔 내 레포/AI브랜치 따로 클론해서 실험해보면 되겠어

File: AI/consumption-main.py [hunk 2]
```diff
@@ -27,10 +27,10 @@ def load_current(BASEDIR_PATH: str) -> list:
 def load_predict(BASEDIR_PATH: str) -> list:
     # -- predict -- #
     x_test = weather_asos.preprocess_test(BASEDIR_PATH)
-    _, predicted = predict_xgboost.run(BASEDIR_PATH, x_test)
+    xgbr, predicted = predict_xgboost.run(BASEDIR_PATH, x_test)
 
     predicted_list: list = predicted.tolist()
-    return predicted_list
+    return xgbr, predicted_list
```
Comment (4923) [inline AI/consumption-main.py:33]: 여기서 xgbr 따로 받은건 시각화 진행하던 과정!
결과에 영향 없으니 무시해도 됨

File: AI/consumption-main.py [hunk 3]
```diff
@@ -86,9 +87,9 @@ def predict_consumption():
     print(json.dumps(consumption))
 
     # --- Visualization ---
-    y_valid = load_2021(BASEDIR_PATH)
-    pred = pd.Series(predict_list)
-    predict_xgboost.visual_xgboost(y_valid, pred)
+    # y_valid = load_2021(BASEDIR_PATH)
+    # pred = pd.Series(predict_list)
+    # predict_xgboost.visual_xgboost(y_valid, pred, xgbr)
```
Comment (4923) [inline AI/consumption-main.py:92]: 시각화코드 주석처리했으니 `consumption-main.py` 실행에 영향을 미치지 않음

File: AI/model/consumption/predict_xgboost.py [hunk 4]
```diff
@@ -28,41 +34,34 @@ def train_xgboost(dataset: pd.DataFrame, x_test):
...
     # Lowest RMSE:  7.0607787236388955
 
     # model
-    xgb_reg = XGBRegressor(seed=0)
+    xgb_reg = XGBRegressor(
+        colsample_bytree=0.3,
+        learning_rate=0.05,
+        max_depth=3,
+        n_estimators=100,
+        seed=0)
```
Comment (4923) [inline AI/model/consumption/predict_xgboost.py:50]: GridSearchCV로 확정한 파라미터를 45번째줄에 모델에 입력한 과정
- issue #36 에서 지적해준 부분 여기서 해결했어

File: AI/model/generation/predict_ann.py [hunk 5]
```diff
@@ -92,58 +101,59 @@ def predict(X, model):
...
 
     # train
-    # model = Model1(name="logistic_regression",
-    #                xdim=9, hdim=17, ydim=1).to(device)
+    # model = Model3(name="logistic_regression",
+    #                xdim=9, h1dim=32, h2dim=16, h3dim=8, ydim=1.to(device)
     # model_save(model, x, y)
-
-# %%
+    None
```
Comment (4923) [inline AI/model/generation/predict_ann.py:159]: 마찬가지로 #36 에서 찾아낸 문제 None 반환으로 해결, 삭제해도 관계 없지만 재학습 하려면 파일 단독 실행해야해서 남겨둠

위 내용은 정빈이랑 튜닝, normalize 해제하는 과정 (#35)

File: AI/preprocess/generation/run.py [hunk 6]
```diff
@@ -0,0 +1,80 @@
+# %%
+
+import numpy as np
+import pandas as pd
+import torch
+
+
+# %%
+# AI/power-generation-model.py
```
Comment (4923) [inline AI/preprocess/generation/run.py:9]: `AI/preprocess/generation/run.py` : 전처리 결과를 main으로 가져가기 위한 실행파일

File: BE/src/power-generation/power-generation.service.ts [hunk 7]
```diff
@@ -6,7 +6,7 @@ export class PowerGenerationService {
   private logger = new Logger('PowerGenerationService');
 
   getPowerGeneration() {
-    const process = spawnSync('python3', ['../AI/power-generation-model.py']);
+    const process = spawnSync('python3', ['../AI/power-generation-main.py']);
```
Comment (4923) [inline BE/src/power-generation/power-generation.service.ts:9]: **중요**

파일명 잘못 호출하고 있길래 변경함 @NEM-NE

**Review by 4923 — COMMENTED** [review]
간단하게 봐야할 내용 및 변경지점만 셀프리뷰했으니 comment 위주로 살펴봐봐
@NEM-NE

**Comment by NEM-NE** [comment]
셀프리뷰까지... 굳굳 지금 바로 테스트해볼게!
